srcpkgs/intel-graphics-compiler/template

```diff
@@ -0,0 +1,26 @@
+# Template file for 'intel-graphics-compiler'
+pkgname=intel-graphics-compiler
+version=1.0.3899
+revision=1
+wrksrc=intel-graphics-compiler-igc-${version}
+#create_wrksrc=yes
+archs="i686* x86_64*"
+build_style=cmake
+configure_args="-DIGC_PREFERRED_LLVM_VERSION='10.0.0'"
+#make_build_args=""
+#make_install_args=""
+#conf_files=""
+#make_dirs="/var/log/dir 0755 root root"
+hostmakedepends="bison flex"
+makedepends="clang-tools-extra llvm10 intel-opencl-clang"
+depends=""
+short_desc="Intel Graphics Compiler for OpenCL"
+maintainer="Stefano Ragni <st3r4g@protonmail.com>"
+license="MIT"
+homepage="https://github.com/intel/intel-graphics-compiler"
+distfiles="https://github.com/intel/intel-graphics-compiler/archive/igc-${version}.tar.gz"
+checksum=fcb721a40a88e1c9bbd45281ba003cf519eaf648435896071bc1e8a80ef05e71
+
```

Comment:
Build of `intel-graphics-compiler` uses a hardcoded reference to `/usr/local/lib/libLLVM-10.so` in `${wrksrc}/IGC/CMakeLists.txt`. This is not the correct location on Void, which is `/usr/lib/libLLVM-10.so` for native compilation but will be somewhere else during cross-compilation. The hard-coded reference should be replaced, either by setting `nocross=yes` and replacing `/usr/local/lib` with `/usr/lib` (*e.g.*, in a post_patch() call to `vsed`) or by finding the appropriate directory based on proper cross-compilation logic.

Reply:
This is an upstream issue that will likely be solved on next release (see https://github.com/intel/intel-graphics-compiler/pull/133). Imported that patch for now

Reply:
This one shouldn't be hurt too much by not being cross compilable. Even if @q66 wants to get a new Xe card to run on PPC.

Reply:
didn't even try to cross compile tbh, I thought the hardware supported `i686* x86_64*` only...

Reply:
You can have cross-compilation behavior even on `i686` or `x86_64*` if, for example, you build with `-a x86_64-musl` or `-a i686` on an `x86_64` masterdir. The result is that all of the `makedepends` and compilers will be taken from a cross-compile tree rather than the native `/usr` tree.

Of course, on `x86_64*`, you can get around this with a native masterdir for any of `i686` or `x86_64*`, because the executables for any of those archs will run natively on the hardware.

Reply:
cross compiling has nothing to do with what architectures it runs on, we have a cross-compiler even for x86_64.

Reply:
ok. Cross compilation seems to be supported by reading cmake output, it's just a matter of locating libLLVM. I'll check again when it's fixed upstream